wasm/js/transform_avro/package.json

```diff
@@ -12,12 +12,12 @@
   "author": "",
   "license": "ISC",
   "dependencies": {
-    "@vectorizedio/wasm-api": ">=21.11.3-si-beta5",
+    "@vectorizedio/wasm-api": ">=22.1.1-rc1",
```

Comment:
I think you should be using the wasm-beta release [v21.12.1-wasm-beta1](https://github.com/redpanda-data/redpanda/releases/tag/v21.12.1-wasm-beta1)

Versions older then this are most likely susceptible to observing that header check exception that you observed